gamestonk_terminal/stocks/discovery/financedatabase_view.py

```diff
@@ -64,6 +66,14 @@ def show_equities(
         data = fd.search_products(data, query=" ".join(name), search="long_name")
     if description is not None:
         data = fd.search_products(data, query=" ".join(description), search="summary")
+    if marketcap is not None:
+        marketcap = " ".join(marketcap).title()
+        if marketcap not in ["Small Cap", "Mid Cap", "Large Cap"]:
+            raise ValueError(
+                f"Invalid choice ({marketcap}) for -mc/--marketcap. Choose from 'Small Cap', "
+                f"'Mid Cap' or 'Large Cap'."
+            )
+        data = fd.search_products(data, query=marketcap, search="market_cap")
```

Comment:
Because `" ".join(marketcap).title()` doesn't work if you add options, I added a custom ValueError.